etcd/etcd.go

```diff
@@ -34,14 +37,27 @@ tests.
 package etcd
 
 import (
+	"errors"
 	"github.com/coreos/go-etcd/etcd"
 	"github.com/zalando/skipper/eskip"
+	"log"
+	"net/http"
 	"path"
-	"strings"
 )
 
 const routesPath = "/routes"
 
+// RouteInfo contains a route id, plus the loaded and parsed route or
+// the parse error in case of failure.
+type RouteInfo struct {
+
+	// The route id plus the route data or if parsing was successful.
+	eskip.Route
+
+	// The parsing error if the parsing failed.
+	ParseError error
+}
```

Comment:
I guess this thing shouldn't contain both at the same time, right? Is there an `Either` type in go?